From a86a9150aeee72c37352c5686465c1a04d7ec78b Mon Sep 17 00:00:00 2001
From: Peter Wu <peter@lekensteyn.nl>
Date: Fri, 20 Feb 2015 16:09:06 +0100
Subject: [PATCH] ui/gtk: fix control combos such as ctrl-a, ctrl-c

Ignore any Ctrl combinations to allow pasting to work. Regression
introduced in "gtk: fix crash on Broadway GDK backend".

Change-Id: I3d53db2d224cb604b52081002ff923c8684a8d0a
Reviewed-on: https://code.wireshark.org/review/7276
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
Tested-by: Balint Reczey <balint@balintreczey.hu>
(cherry picked from commit 1a10f5f2b470dede438ac33c25e0cb86c8e12e17)
Reviewed-on: https://code.wireshark.org/review/7277
Reviewed-by: Peter Wu <peter@lekensteyn.nl>
---
 ui/gtk/filter_autocomplete.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/ui/gtk/filter_autocomplete.c b/ui/gtk/filter_autocomplete.c
index c69879d..8d75526 100644
--- a/ui/gtk/filter_autocomplete.c
+++ b/ui/gtk/filter_autocomplete.c
@@ -368,9 +368,17 @@ filter_string_te_key_pressed_cb(GtkWidget *filter_te, GdkEventKey *event, gpoint
   if (!key_string)
     key_string = g_strdup("");
 
-  /* If the pressed key is SHIFT then we have nothing to do with the pressed key. */
-  if( k == GDK_Shift_L || k == GDK_Shift_R)
+  /* Ignore Ctrl combinations (Ctrl-C, Ctrl-V, Ctrl-Right, ...). */
+  if (event->state & GDK_CONTROL_MASK)
     goto exit;
+  /* Pressing modifiers such as Shift or Ctrl should not hide the list. */
+  switch (k) {
+  case GDK_Shift_L:
+  case GDK_Shift_R:
+  case GDK_KEY_Control_L:
+  case GDK_KEY_Control_R:
+    goto exit;
+  }
 
   if (popup_win)
     gtk_widget_show(popup_win);
-- 
2.1.4

